### PR TITLE
FE BL view comments

### PR DESCRIPTION
**Pull request by didOstap**

### CODE REVIEW

File: client/src/components/Comments/index.tsx [hunk 1]
```diff
@@ -1,34 +1,103 @@
...
-                <div className='one-comment px-3' key={comment.id}>
-                    <img
-                        src={comment.image || noAvatar}
-                        alt='avatar'
-                        width={24}
-                        height={24}
-                        className='img-fluid rounded-circle mt-1 mr-1 mb-1'
+interface ILocalState {
+    postId: string;
+    comments: [];
```
Comment (InnaKutsik) [inline client/src/components/Comments/index.tsx]: [] of what?

File: client/src/components/Comments/index.tsx [hunk 2]
```diff
@@ -1,34 +1,103 @@
...
+interface IState {
+    comments: ILocalState;
+    ownProps: {
+        postId: string;
+    };
+}
+
+interface IComments {
+    postId: string;
+    comments: [];
```
Comment (InnaKutsik) [inline client/src/components/Comments/index.tsx]: [] of what?

File: client/src/components/Comments/index.tsx [hunk 3]
```diff
@@ -1,34 +1,103 @@
...
+    commentsPage: number;
+    commentsLoading: boolean;
+    allCommentsLoaded: boolean;
+    getComments: (postId: string) => void;
+    getMoreComments: (postId: string, page: number) => void;
+    addNextCommentsPage: (commentsPage: number) => void;
+    resetComments: () => void;
+}
```
Comment (InnaKutsik) [inline client/src/components/Comments/index.tsx]:
```
interface IOwnCommentsProps {
    commentsLoading: boolean;
    getComments: (postId: string) => void;
    getMoreComments: (postId: string, page: number) => void;
    addNextCommentsPage: (commentsPage: number) => void;
    resetComments: () => void;
}

export type ICommentsProps = ILocalState & IOwnCommentsProps ;

File: client/src/components/Comments/index.tsx (hunk 1 above)
Comment (didOstap) [reply client/src/components/Comments/index.tsx]: fixed

File: client/src/components/Comments/index.tsx (hunk 2 above)
Comment (didOstap) [reply client/src/components/Comments/index.tsx]: fixed

File: client/src/components/Comments/index.tsx (hunk 3 above)
Comment (didOstap) [reply client/src/components/Comments/index.tsx]: fixed

File: client/src/store/comments/actions.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            if (!res.data.commentsAll.length) {
+                dispatch(allCommentsLoaded());
+                return;
+            }
+
+            dispatch(getMoreCommentsSuccess(res.data.commentsAll));
+        } catch (e) {
+            dispatch(showAlert(e.response.data.message, 'danger'));
+        }
+    };
```
Comment (InnaKutsik) [inline client/src/store/comments/actions.ts]: why not to make one function?
```
export const getComments = (postId: string, page: number = 1): (dispatch: Dispatch) => Promise<void> =>
    async (dispatch: Dispatch): Promise<void> => {
        try {
            dispatch(getCommentsPending());
            const res = await AuthAPI.get(`comments/${postId}/${page}`);

            if (!res.data.commentsAll.length) {
                dispatch(allCommentsLoaded());
                return;
            }

            const successAction = page === FIRST_PAGE ? getCommentsSuccess : getMoreCommentsSuccess;
             
             dispatch(successAction(res.data.commentsAll));
        } catch (e) {
            dispatch(showAlert(e.response.data.message, 'danger'));
        }
    };

File: client/src/components/Comments/index.tsx [hunk 5]
```diff
@@ -1,34 +1,101 @@
...
+                                width={24}
+                                height={24}
+                                className='img-fluid rounded-circle mt-1 mr-1 mb-1'
+                            />
+                            <span className='mt-1'>{comment.authorId.username}</span>
+                            <div className='d-inline-flex mt-3 float-right edit-delete-comment'>
+                                <i className='fa fa-pencil mr-2 edit-comment'/>
+                                <i className='fa fa-trash-o delete-comment'/>
+                            </div>
+                            <p>{comment.comment}</p>
```
Comment (InnaKutsik) [inline client/src/components/Comments/index.tsx]: it seems to me that someone wrote component for it. Pls check

File: client/src/store/comments/actions.ts (hunk 4 above)
Comment (didOstap) [reply client/src/store/comments/actions.ts]: fixed

File: client/src/components/Comments/index.tsx (hunk 5 above)
Comment (didOstap) [reply client/src/components/Comments/index.tsx]: Checked
No another component for it

**Pull request merged**